### PR TITLE
Dexsearch: minor improvements

### PR DESCRIPTION
**Pull request by CheeseMuffin**
This makes it so you can do `/ds 1, normal` instead of having to type `/ds gen1, normal type`
For types and egg groups with the same name, types take priority, but you can still do `/ds fairy group` if necessary. Similar changes to `/ms`

Mostly did this bc Lady Monita lets you do these same things

### CODE REVIEW

**Comment by TheImmortal** [comment]
imo it should be part of the command, ``/ds1 tackle`` or ``/ds2 tackle``

**Comment by CheeseMuffin** [comment]
That doesn't really seem necessary, just an extra space between the `ds` and `1`...

**Comment by Zarel** [comment]
I don't think there's a need for `/ds1` etc.

**Comment by Zarel** [comment]
I like `normal` instead of `normal type`, but I don't like `1` instead of `gen1`. I think filtering by gen is rare enough that typing "gen" is clearer and better.

**Comment by CheeseMuffin** [comment]
Is there a big disadvantage to having it? I use `ds` quite a bit for gens and it'd be nice to be able to just type the number instead (especially because Lady Monita lets you just use the number).

**Comment by Zarel** [comment]
Maybe `g1`? It's mostly because numbers can be ambiguous.

**Comment by CheeseMuffin** [comment]
That makes sense, updated

**Comment by TheImmortal** [comment]
I've just been informed that the gen parameter searches for Pokemon from that gen only. Which isn't what I thought. In that case, my suggestion doesn't fit.

I do want to mention however that it is really dumb that you can search for a Pokemon from a specific gen rather than search a specific gen for stuff, which is what I thought it did. How is that useful at all? Compared to being able to use dexsearch to search stuff in specific gens. If you can, please implement past gen ds, and the command can be what I suggested.

**Comment by CheeseMuffin** [comment]
Does something like `/ds gen < 5` sound good (and then add an alias for `ds5`)?

**Comment by CheeseMuffin** [comment]
Sorry I have no idea what I did to the commits, but added the `/ds5` and such that looks for everything from gen 5 or earlier

**Comment by Zarel** [comment]
Side note: Commits should most likely not contain the word "Travis".

Like, if I tell you your code is bad, your commit shouldn't say "Fix Zarel", and if Travis tells you your code has a build error, your commit shouldn't say "Fix Travis". You can just say "Fix build error" (or, in practice, use an amend/fixup commit).

File: chat-plugins/datasearch.js [hunk 1]
```diff
@@ -85,11 +85,19 @@ const PM = exports.PM = new DatasearchManager({
...
+	ds3: 'dexsearch',
+	ds4: 'dexsearch',
+	ds5: 'dexsearch',
+	ds6: 'dexsearch',
+	ds7: 'dexsearch',
 	dsearch: 'dexsearch',
 	dexsearch: function (target, room, user, connection, cmd, message) {
 		if (!this.canBroadcast()) return;
+		let targetGen = parseInt(cmd[cmd.length - 1]);
+		if (targetGen) target += ", gen<" + (targetGen + 1);
```
Comment (Zarel) [inline chat-plugins/datasearch.js]: This doesn't just apply to, like, gen range, you know. It also affects learnsets, stats, abilities...

**Comment by CheeseMuffin** [comment]
This is the first time using the Dex.mod stuff so hopefully I didn't screw it up, but now it should use the info from that gen.

**Comment by Zarel** [comment]
someone review?

**Comment by CheeseMuffin** [comment]
Rewrote the `ms` code to allow | as a separation, so its more similar to `ds`

**Pull request merged**